nupic/regions/CLAClassifierRegion.py

```diff
@@ -214,30 +258,70 @@ def setParameter(self, name, index, value):
     else:
       return PyRegion.setParameter(self, name, index, value)
 
-
   def reset(self):
     pass
 
-
   def compute(self, inputs, outputs):
     """
     Process one input sample.
     This method is called by the runtime engine.
 
-    We don't use this method in this region because the inputs and outputs don't
-    fit the standard "vector of reals" used by the engine. Instead, call
-    the customCompute() method directly
     """
 
-    pass
-
+    # An input can potentially belong to multiple categories. 
+    # If a category value is < 0, it means that the input does not belong to 
+    # that category.
+    categories = [category for category in inputs["categoryIn"]
+                  if category >= 0]
+
+    activeCells = inputs["bottomUpIn"]
+    patternNZ = activeCells.nonzero()[0]
+
+    # Allow to train on multiple input categories. 
+    # Do inference first, and then train on all input categories.
+    #   1. Call classifier. Don't train. Just inference. Train after.
+    clResults = self._claClassifier.compute(recordNum=self.recordNum,
+                                            patternNZ=patternNZ,
+                                            classification=None,
+                                            learn=False,
+                                            infer=self.inferenceMode)
+
+    for category in categories:
+      classificationIn = {
+        "bucketIdx": int(category),
+        "actValue": int(category)
+        }
+
+      #   2. Train classifier, no inference
+      self._claClassifier.compute(recordNum=self.recordNum,
+                                  patternNZ=patternNZ,
+                                  classification=classificationIn,
+                                  learn=self.learningMode,
+                                  infer=False)
+
+    outputs['actualValues'] = clResults["actualValues"]
+    for step in self.stepsList:
+      stepIndex = self.stepsList.index(step)
+      categoryOut = clResults["actualValues"][clResults[step].argmax()]
+      outputs['categoriesOut'][stepIndex] = categoryOut
+
+      # Flaten the rest of the output. For example:
```

Comment:
I like this comment. It kind of suggests that yo ucould move this logic into a function though which would make it more testable.